### PR TITLE
Sharable Select component

### PR DESCRIPTION
**Pull request by fionaphillips**
Resolves [CG12](https://unosquarebelfast.atlassian.net/secure/RapidBoard.jspa?rapidView=12&projectKey=CG&modal=detail&selectedIssue=CG-12)

There is a bug in react-select that causes the top option to always be in focus. Ticket to fix when the next release goes out - https://unosquarebelfast.atlassian.net/browse/CG-27

**Mobile View**
selectComponentMobileView

### CODE REVIEW

File: src/__tests__/components/common/Select/index.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import context from 'jest-context';
+import Select from 'react-select';
+import 'SetupTests';
```
Comment (BrownMark) [inline src/__tests__/components/common/Select/index.jsx]: you should need to include setuptests in each test file. 
```
"jest": {
    "setupTestFrameworkScriptFile": "./setupTests.js",

```

Adding that to package.json should work

**Review by BrownMark — CHANGES_REQUESTED** [review]
One small thing, but looking really well :)

File: src/components/common/Label/index.jsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { StyledLabel } from './styled';
 
 const Label = ({ text, isRequired }) => (
   <StyledLabel>
-    {text} {isRequired && <div>*</div>}
+    {text} {isRequired && <span>*</span>}
```
Comment (girwin90) [inline src/components/common/Label/index.jsx:7]: Nice

File: src/__tests__/components/common/Select/index.jsx (hunk 1 above)
Comment (BrownMark) [reply src/__tests__/components/common/Select/index.jsx]: or possibly the jest.config.json

File: src/components/common/Select/index.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          value={selected}
+        />
+        {/* placeholder until component is hooked up to API */}
+        <div className="selectedDiv">Selected option: {selected && selected.label}</div>
+      </StyledSelect>
+    );
+  }
+}
+
+export default SelectComponent;
```
Comment (girwin90) [inline src/components/common/Select/index.jsx]: A nit pick, sorry! I prefer if the export is at the bottom of the file. If you disagree, happy to discuss!

**Review by girwin90 — APPROVED** [review]
Minor comments. Still approving. Good graft, following the pattern and it all looks tidy. Nice going.

File: src/components/common/Select/index.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  text,
+  isRequired,
+  name,
+  isSearchable,
+  options,
+  isDisabled,
+}) => {
+  console.log('options', options);
+
+  const triggerOnChange = e => {
```
Comment (BrownMark) [inline src/components/common/Select/index.jsx]: is this needed? Might be able to remove it and just handleOnChange passed as a prop.

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: src/components/common/Select/container.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { PropTypes as PT } from 'prop-types';
+
+const SelectContainer = Wrapped =>
+  class Container extends React.Component {
```
Comment (BrownMark) [inline src/components/common/Select/container.jsx]: don't need to add Container here. 

```
class extends React.Component {
```

File: src/components/common/Select/container.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { PropTypes as PT } from 'prop-types';
+
+const SelectContainer = Wrapped =>
```
Comment (BrownMark) [inline src/components/common/Select/container.jsx]: this can be

```
export default Wrapped =>
```

**Review by BrownMark — CHANGES_REQUESTED** [review]
just 2 super minor things

File: src/components/common/Select/container.jsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    state = { selected: null };
+
+    handleOnChange = selected => {
+      this.setState({ selected });
+    };
+
+    render() {
+      const { selected } = this.state;
+      console.log('selected: ', selected);
```
Comment (girwin90) [inline src/components/common/Select/container.jsx]: A log is here?

File: src/components/common/Select/index.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  selected,
+  handleOnChange,
+  text,
+  isRequired,
+  name,
+  isSearchable,
+  options,
+  isDisabled,
+}) => {
+  console.log('options', options);
```
Comment (girwin90) [inline src/components/common/Select/index.jsx]: Debug here.

File: src/components/common/Select/index.jsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        onChange={triggerOnChange}
+        value={selected || null}
+      />
+      {/* placeholder until component is hooked up to API */}
+      <div className="selectedDiv">Selected option: {selected && selected.label}</div>
+    </StyledSelect>
+  );
+};
+
+export default SelectContainer(SelectComponent);
```
Comment (girwin90) [inline src/components/common/Select/index.jsx]: I'd drop this below the prop types.

**Review by girwin90 — CHANGES_REQUESTED** [review]

File: src/__tests__/components/common/Select/index.jsx (hunk 1 above)
Comment (fionaphillips) [reply src/__tests__/components/common/Select/index.jsx]: I created an alias and imported this on line 5

**Review by BrownMark — APPROVED** [review]

**Pull request merged**